specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/applicationGateway.json

```diff
@@ -2642,13 +2642,36 @@
           "type": "string",
           "description": "Header name of the header configuration."
         },
+        "headerValueMatcher": {
+          "type": "object",
+          "$ref": "#/definitions/HeaderValueMatcher",
+          "description": "An optional field under \"Rewrite Action\". It lets you capture and modify the value(s) of a specific header when multiple headers with the same name exist. Currently supported for Set-Cookie Response header only. For more details, visit https://aka.ms/appgwheadercrud"
+        },
         "headerValue": {
           "type": "string",
           "description": "Header value of the header configuration."
         }
       },
       "description": "Header configuration of the Actions set in Application Gateway."
     },
+    "HeaderValueMatcher": {
+      "type": "object",
+      "properties": {
+        "pattern": {
+          "type": "string",
+          "description": "The pattern, either fixed string or regular expression, that evaluates if a header value should be selected for rewrite."
+        },
+        "ignoreCase": {
+          "type": "boolean",
+          "description": "Setting this parameter to truth value with force the pattern to do a case in-sensitive comparison."
+        },
+        "negate": {
+          "type": "boolean",
+          "description": "Setting this value as truth will force to check the negation of the condition given by the user in the pattern field."
```

Comment:
consider updating these to enums. This is not a good model leading to proliferation of too many boolean switches and toggles inthe API.

[https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md](https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md)

Reply:
Both the "ignoreCase" and "negate" properties are really boolean and there are no possible values for both other than true and false.

According to the shared link: https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md => if the property is really boolean in nature, use of enum can be skipped.

Reply:
Hi @raosuhas could you kindly review my previous comment and indicate if there are any outstanding tasks for me?